### PR TITLE
Refactor get_report_render_url

### PR DESCRIPTION
**Pull request by benrudolph**
This passes data as an object for a POST so date filter is correctly
read. The issue for this was that since the arguments were being passed in the URL django was putting them in the GET request object. The datefilter decorator checks the method and then grabs the corresponding request dict therefore the dates weren't being read. See this line of code:

https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/decorators/datespan.py#L30

http://manage.dimagi.com/default.asp?159628#887826

### CODE REVIEW

File: corehq/apps/reports/static/reports/javascripts/reports.config.js [hunk 1]
```diff
@@ -162,8 +163,29 @@ var HQReport = function (options) {
                     "&enddate="+self.datespan.enddate;
             }
         }
-        return window.location.pathname.replace(self.urlRoot,
-            self.urlRoot+render_type+"/")+"?"+params + (additionalParams == undefined ? "" : "&" + additionalParams);
+        params += (additionalParams == undefined ? "" : "&" + additionalParams);
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/reports.config.js]: Use '===' to compare with 'undefined'.

File: corehq/apps/reports/static/reports/javascripts/reports.config.js [hunk 2]
```diff
@@ -162,8 +163,29 @@ var HQReport = function (options) {
...
+                    .replace(/=/g,'":"') +
+                '"}');
+
+        }
+        return params;
+
+    }
+
+    function getReportBaseUrl(renderType) {
+        return window.location.pathname.replace(self.urlRoot, self.urlRoot+renderType+"/")
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/reports.config.js]: Missing semicolon.

File: corehq/apps/reports/static/reports/javascripts/reports.config.js [hunk 3]
```diff
@@ -43,14 +43,15 @@ var HQReport = function (options) {
                         e.preventDefault();
                         if (self.isExportAll) {
                             $.ajax({
-                                url: get_report_render_url("export"),
+                                url: getReportBaseUrl("export"),
+                                data: getReportParams(undefined, true),
```
Comment (esoergel) [inline corehq/apps/reports/static/reports/javascripts/reports.config.js]: I think the view receiving this request will see it the same either way, but did you confirm that it still parses the request params correctly?

Comment (esoergel) [inline corehq/apps/reports/static/reports/javascripts/reports.config.js]: <offlined>

**Pull request merged**